common/lib/transport/comettransport.js

```diff
@@ -216,18 +216,25 @@ var CometTransport = (function() {
 			if(err) Logger.logAction(Logger.LOG_ERROR, 'CometTransport.sendItems()', 'on complete: err = ' + Utils.inspectError(err));
 			self.sendRequest = null;
 
-			/* the results of the request usually get handled as protocol responses instead of send errors */
+			/* the result of the request, even if a nack, is usually a protocol response
+			 * contained in the data. An err is anomolous, and indicates some issue with the
+			 * network,transport, or connection */
+			if(err) {
+				if(err.code) {
```

Comment:
I can't think of a better way to check this, but it's not ideal doing a duck-typing check this way. I take it because JSONP we can't check for `X-Ably-xxx` headers?

Edit: I appreciate there's a TODO also here and I think there is an issue for that already.

Reply:
you're right, it is a bit hacky. I made a stab at a realtime PR to actually fix this (by always returning ProtocolMessage responses to comet requests, rather than sometimes those and sometimes REST-style error responses) back in March: https://github.com/ably/realtime/pull/2353/files . But it got parked, as it turned out to be more complex than I was expecting and other things took priority since, while hacky, the current way does mostly just work